src/solana/stake_account.ts

```diff
@@ -0,0 +1,162 @@
+// Copied from https://github.com/igneous-labs/solana-stake-sdk
+import { AccountInfo, PublicKey, ParsedAccountData } from '@solana/web3.js';
+import {
+  coerce,
+  create,
+  enums,
+  Infer,
+  instance,
+  nullable,
+  number,
+  string,
+  type,
+} from 'superstruct';
+import BigNumber from 'bignumber.js';
+
+export const PublicKeyFromString = coerce(
+  instance(PublicKey),
+  string(),
+  (value) => new PublicKey(value),
+);
+
+export class ParseStakeAccountError extends Error {}
+
+export const StakeState = {
+  Inactive: 'inactive',
+  Activating: 'activating',
+  Active: 'active',
+  Deactivating: 'deactivating',
+  Deactivated: 'deactivated',
+};
+
+export const StakeAccountType = enums([
+  'uninitialized',
+  'initialized',
+  'delegated',
+  'rewardsPool',
+]);
+
+export const BigNumFromString = coerce(
+  instance(BigNumber),
+  string(),
+  (value) => {
+    if (typeof value === 'string') return new BigNumber(value, 10);
+    throw new Error('invalid big num');
+  },
+);
+
+export type StakeMeta = Infer<typeof StakeMeta>;
```

Comment:
it's better to move types into separate file.

Reply:
@Messer4  should we move stake account types into solana common types ?